Verification test should check cluster Ready state before retrieving kubeconfig

## Problem

The verification test `TestVerification_RetrieveKubeconfig` in `test/06_verification_test.go` attempts to retrieve the kubeconfig secret regardless of the cluster's provisioning status. This leads to confusing failures when the cluster is still provisioning.

### Current Behavior

The test:
1. Immediately tries to retrieve the kubeconfig secret
2. Finds the secret exists but has an empty `value` field
3. Fails with "Secret value is empty, cannot decode kubeconfig"

### Root Cause

When a cluster is still in `Provisioning` phase:
- The kubeconfig secret is created by ASO but with an empty value
- ASO only populates the secret after the HcpOpenShiftCluster resource is fully provisioned in Azure
- The test doesn't check cluster status before attempting retrieval

### Example Failure

```
=== RUN   TestVerification_RetrieveKubeconfig
    06_verification_test.go:64: Secret value is empty, cannot decode kubeconfig
--- FAIL: TestVerification_RetrieveKubeconfig (0.16s)
```

With cluster status showing:
```
NAME        PHASE          AGE
rcap-stage  Provisioning   46m
```

## Proposed Solution

Add a precondition check at the start of `TestVerification_RetrieveKubeconfig` to verify the cluster is in `Ready` state before attempting to retrieve the kubeconfig.

```go
func TestVerification_RetrieveKubeconfig(t *testing.T) {
    config := NewTestConfig()
    
    // Check cluster is ready before attempting kubeconfig retrieval
    clusterPhase := getClusterPhase(t, config)
    if clusterPhase != "Provisioned" {
        t.Skipf("Cluster is not ready (current phase: %s), skipping kubeconfig retrieval", clusterPhase)
    }
    
    // ... existing kubeconfig retrieval logic
}
```

Alternatively, the test could wait for the cluster to become ready with a timeout, similar to how `TestDeployment_WaitForControlPlane` works.

## Acceptance Criteria

- [ ] Test checks cluster phase before attempting kubeconfig retrieval
- [ ] Test skips with clear message if cluster is not ready
- [ ] Error message indicates current cluster phase for debugging

## Related

- `test/06_verification_test.go:24-64`
- `test/05_deploy_crs_test.go` - deployment phase that should complete first

--- a/test/06_verification_test.go
+++ b/test/06_verification_test.go
@@ -26,6 +26,19 @@ func TestVerification_RetrieveKubeconfig(t *testing.T) {
 	// Use the provisioned cluster name from aro.yaml
 	provisionedClusterName := config.GetProvisionedClusterName()
 
+	// Check cluster phase before attempting kubeconfig retrieval (fixes #275)
+	// When a cluster is still provisioning, ASO creates the kubeconfig secret with an empty
+	// value, which causes confusing "Secret value is empty" errors.
+	clusterPhase, err := GetClusterPhase(t, context, provisionedClusterName)
+	if err != nil {
+		t.Skipf("Cannot determine cluster phase: %v (cluster resource may not exist yet)", err)
+	}
+
+	if clusterPhase != ClusterPhaseProvisioned {
+		t.Skipf("Cluster is not ready (current phase: %s), skipping kubeconfig retrieval. "+
+			"Wait for cluster provisioning to complete or run TestDeployment_WaitForControlPlane first.", clusterPhase)
+	}
+
 	// Kubeconfig output path - use helper for consistency
 	kubeconfigPath := getKubeconfigPath(config)
 

--- a/test/helpers.go
+++ b/test/helpers.go
@@ -856,6 +856,127 @@ func isRetryableKubectlError(output string, err error) bool {
 	return false
 }
 
+// GetClusterPhase retrieves the current phase of a CAPI Cluster resource.
+// Returns the phase string (e.g., "Provisioning", "Provisioned", "Failed") or an error.
+// This is useful for checking if a cluster is ready before attempting operations that
+// require the cluster to be fully provisioned (like retrieving kubeconfig).
+//
+// Parameters:
+//   - t: testing context
+//   - kubeContext: kubectl context to use (e.g., "kind-capz-tests-stage")
+//   - clusterName: name of the Cluster resource to check
+//
+// Returns the phase string or an error if the cluster is not found or the phase cannot be retrieved.
+func GetClusterPhase(t *testing.T, kubeContext, clusterName string) (string, error) {
+	t.Helper()
+
+	output, err := RunCommandQuiet(t, "kubectl", "--context", kubeContext, "get", "cluster",
+		clusterName, "-o", "jsonpath={.status.phase}")
+	if err != nil {
+		return "", fmt.Errorf("failed to get cluster phase: %w", err)
+	}
+
+	phase := strings.TrimSpace(output)
+	if phase == "" {
+		return "", fmt.Errorf("cluster phase is empty (cluster may not have status yet)")
+	}
+
+	return phase, nil
+}
+
+// ClusterPhaseProvisioned is the phase value indicating a cluster is fully provisioned and ready.
+const ClusterPhaseProvisioned = "Provisioned"
+
+// ClusterPhaseProvisioning is the phase value indicating a cluster is still being provisioned.
+const ClusterPhaseProvisioning = "Provisioning"
+
+// ClusterPhaseFailed is the phase value indicating a cluster provisioning has failed.
+const ClusterPhaseFailed = "Failed"
+
+// IsClusterReady checks if a cluster is in the Provisioned phase.
+// Returns true if the cluster is ready, false otherwise.
+func IsClusterReady(t *testing.T, kubeContext, clusterName string) bool {
+	t.Helper()
+
+	phase, err := GetClusterPhase(t, kubeContext, clusterName)
+	if err != nil {
+		return false
+	}
+
+	return phase == ClusterPhaseProvisioned
+}
+
+// DefaultClusterReadyTimeout is the default timeout for waiting for a cluster to become ready.
+const DefaultClusterReadyTimeout = 60 * time.Minute
+
+// DefaultClusterReadyPollInterval is the default interval between cluster ready checks.
+const DefaultClusterReadyPollInterval = 30 * time.Second
+
+// WaitForClusterReady waits for a cluster to reach the Provisioned phase.
+// This is useful when you need to wait for a cluster to be fully provisioned before
+// performing operations that require the cluster to be ready (like retrieving kubeconfig).
+//
+// Parameters:
+//   - t: testing context
+//   - kubeContext: kubectl context to use (e.g., "kind-capz-tests-stage")
+//   - clusterName: name of the Cluster resource to check
+//   - timeout: maximum time to wait for the cluster to become ready (use 0 for default of 60m)
+//
+// Returns nil if the cluster becomes ready, or an error if the timeout is reached or the cluster fails.
+func WaitForClusterReady(t *testing.T, kubeContext, clusterName string, timeout time.Duration) error {
+	t.Helper()
+
+	if timeout == 0 {
+		timeout = DefaultClusterReadyTimeout
+	}
+
+	pollInterval := DefaultClusterReadyPollInterval
+	startTime := time.Now()
+
+	PrintToTTY("\n=== Waiting for cluster to be ready ===\n")
+	PrintToTTY("Cluster: %s | Timeout: %v | Poll interval: %v\n\n", clusterName, timeout, pollInterval)
+	t.Logf("Waiting for cluster '%s' to be ready (timeout: %v)...", clusterName, timeout)
+
+	iteration := 0
+	for {
+		elapsed := time.Since(startTime)
+		remaining := timeout - elapsed
+
+		if elapsed > timeout {
+			PrintToTTY("\n❌ Timeout waiting for cluster to be ready after %v\n\n", elapsed.Round(time.Second))
+			return fmt.Errorf("timeout waiting for cluster '%s' to be ready after %v", clusterName, elapsed.Round(time.Second))
+		}
+
+		iteration++
+
+		PrintToTTY("[%d] Checking cluster phase...\n", iteration)
+
+		phase, err := GetClusterPhase(t, kubeContext, clusterName)
+		if err != nil {
+			PrintToTTY("[%d] ⚠️  Failed to get cluster phase: %v\n", iteration, err)
+			t.Logf("Failed to get cluster phase (iteration %d): %v", iteration, err)
+		} else {
+			PrintToTTY("[%d] 📊 Cluster phase: %s\n", iteration, phase)
+			t.Logf("Cluster phase (iteration %d): %s", iteration, phase)
+
+			switch phase {
+			case ClusterPhaseProvisioned:
+				PrintToTTY("\n✅ Cluster is ready! (took %v)\n\n", elapsed.Round(time.Second))
+				t.Logf("Cluster '%s' is ready (took %v)", clusterName, elapsed.Round(time.Second))
+				return nil
+			case ClusterPhaseFailed:
+				PrintToTTY("\n❌ Cluster provisioning failed!\n\n")
+				return fmt.Errorf("cluster '%s' provisioning failed", clusterName)
+			}
+		}
+
+		// Report progress
+		ReportProgress(t, iteration, elapsed, remaining, timeout)
+
+		time.Sleep(pollInterval)
+	}
+}
+
 // ValidateYAMLFile validates that a file contains valid YAML.
 // Returns an error if the file is empty, unreadable, or contains invalid YAML syntax.
 // This is more robust than just checking file size, as it verifies YAML structure.

--- a/test/helpers_test.go
+++ b/test/helpers_test.go
@@ -1243,3 +1243,35 @@ func TestRetryConstants(t *testing.T) {
 		t.Errorf("DefaultApplyRetryDelay = %v, expected at least 5s", DefaultApplyRetryDelay)
 	}
 }
+
+func TestClusterPhaseConstants(t *testing.T) {
+	// Verify cluster phase constants are correctly defined
+	if ClusterPhaseProvisioned != "Provisioned" {
+		t.Errorf("ClusterPhaseProvisioned = %q, expected \"Provisioned\"", ClusterPhaseProvisioned)
+	}
+
+	if ClusterPhaseProvisioning != "Provisioning" {
+		t.Errorf("ClusterPhaseProvisioning = %q, expected \"Provisioning\"", ClusterPhaseProvisioning)
+	}
+
+	if ClusterPhaseFailed != "Failed" {
+		t.Errorf("ClusterPhaseFailed = %q, expected \"Failed\"", ClusterPhaseFailed)
+	}
+}
+
+func TestClusterReadyConstants(t *testing.T) {
+	// Verify cluster ready timeout constants have sensible values
+	if DefaultClusterReadyTimeout < 30*time.Minute {
+		t.Errorf("DefaultClusterReadyTimeout = %v, expected at least 30m", DefaultClusterReadyTimeout)
+	}
+
+	if DefaultClusterReadyPollInterval < 10*time.Second {
+		t.Errorf("DefaultClusterReadyPollInterval = %v, expected at least 10s", DefaultClusterReadyPollInterval)
+	}
+
+	// Verify poll interval is less than timeout
+	if DefaultClusterReadyPollInterval >= DefaultClusterReadyTimeout {
+		t.Errorf("DefaultClusterReadyPollInterval (%v) should be less than DefaultClusterReadyTimeout (%v)",
+			DefaultClusterReadyPollInterval, DefaultClusterReadyTimeout)
+	}
+}

